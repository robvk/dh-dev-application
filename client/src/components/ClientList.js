import React from "react";
import map from "lodash/map";

import "./ClientList.css";
import ClientListItem from "./ClientListItem";
import NoResult from "./NoResult";
import type { Client } from "../lib/flowTypes";

type Props = {
  clients: Array<Client>,
  selectedClient: Client
};

const ClientList = (props: Props) => {
  const { clients, selectedClient } = props;

  return (
    <div className="client-dropdown-list">
      {clients && clients.length > 0 ? (
        map(clients, client => (
          <ClientListItem
            key={client.id}
            client={client}
            isSelected={selectedClient && client.id === selectedClient.id}
          />
        ))
      ) : (
        <NoResult />
      )}
    </div>
  );
};

export default ClientList;
