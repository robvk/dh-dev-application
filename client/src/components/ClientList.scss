@import "../appColors";

.client-dropdown-list {
  z-index: 1;
  width: 100%;
  border: 1px solid $primary-color;
  border-top: 0;
  box-sizing: border-box;
}

.client-dropdown-list__item {
  padding: 10px;
}

.client-dropdown-list__item:hover {
  background-color: $primary-color;
  color: $background-color;
}
