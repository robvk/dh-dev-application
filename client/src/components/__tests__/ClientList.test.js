import React from "react";
import ReactDOM from "react-dom";

import ClientList from "../ClientList";

const testProps = {
  clients: [
    {
      id: "59a2afbbfc13ae278700044c",
      first_name: "Rochella",
      last_name: "Chipchase",
      email: "rchipchase0@joomla.org",
      gender: "Female",
      photo:
        "https://robohash.org/molestiaedoloremquesed.jpg?size=50x50&set=set1",
      origin: "Egypt"
    },
    {
      id: "59a2afbbfc13ae278700044d",
      first_name: "Sophia",
      last_name: "Mayler",
      email: "smayler1@goo.ne.jp",
      gender: "Female",
      photo:
        "https://robohash.org/perspiciatisoditatque.png?size=50x50&set=set1",
      origin: "Zimbabwe"
    },
    {
      id: "59a2afbbfc13ae278700044e",
      first_name: "Bartram",
      last_name: "Nassi",
      email: "bnassi2@goodreads.com",
      gender: "Male",
      photo: "https://robohash.org/aeablanditiis.bmp?size=50x50&set=set1",
      origin: "Egypt"
    }
  ]
};

it("renders correctly", () => {
  const wrapper = shallow(<ClientList />);
  wrapper.setProps(testProps);
  expect(wrapper).toMatchSnapshot();
});

it("has a child for every client", () => {
  const wrapper = shallow(<ClientList />);
  wrapper.setProps(testProps);
  expect(wrapper.find("ClientListItem").length).toBe(3);
});
