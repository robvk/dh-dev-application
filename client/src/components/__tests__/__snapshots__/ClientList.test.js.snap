// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="client-dropdown-list"
>
  <ClientListItem
    client={
      Object {
        "email": "rchipchase0@joomla.org",
        "first_name": "Rochella",
        "gender": "Female",
        "id": "59a2afbbfc13ae278700044c",
        "last_name": "Chipchase",
        "origin": "Egypt",
        "photo": "https://robohash.org/molestiaedoloremquesed.jpg?size=50x50&set=set1",
      }
    }
    key="59a2afbbfc13ae278700044c"
  />
  <ClientListItem
    client={
      Object {
        "email": "smayler1@goo.ne.jp",
        "first_name": "Sophia",
        "gender": "Female",
        "id": "59a2afbbfc13ae278700044d",
        "last_name": "Mayler",
        "origin": "Zimbabwe",
        "photo": "https://robohash.org/perspiciatisoditatque.png?size=50x50&set=set1",
      }
    }
    key="59a2afbbfc13ae278700044d"
  />
  <ClientListItem
    client={
      Object {
        "email": "bnassi2@goodreads.com",
        "first_name": "Bartram",
        "gender": "Male",
        "id": "59a2afbbfc13ae278700044e",
        "last_name": "Nassi",
        "origin": "Egypt",
        "photo": "https://robohash.org/aeablanditiis.bmp?size=50x50&set=set1",
      }
    }
    key="59a2afbbfc13ae278700044e"
  />
</div>
`;

exports[`renders correctly when no result 1`] = `
<div
  className="client-dropdown-list"
>
  <NoResult />
</div>
`;
